terraform/lib/dependabot/terraform/file_updater.rb

```diff
@@ -47,7 +47,7 @@ def updated_terraform_file_content(file)
           case new_req[:source][:type]
           when "git"
             update_git_declaration(new_req, old_req, content, file.name)
-          when "registry"
+          when "registry", "provider"
```

Comment:
The naming of these still bugs me, they should probably be `registry_module` and `registry_provider`

Reply:
It's a bit unclear that `registry` refers to modules. I'm not sure what the impact vs. effort is to address this. Perhaps, if it's important enough we can consider it in a follow up PR. WDYT?

Reply:
Agreed!